src/components/Alarm/Configurator/AlarmConfigurator.tsx

```diff
@@ -145,55 +129,26 @@ const AlarmConfig: FC<Props> = (props: Props) => {
    */
   const restoreAlarmConfig = () => {
     const savedAlarmConfigStr = localStorage.getItem(ALARM_DOC_NAME);
-    const initFormFields: FormFields = cloneDeep(defaultFormfields);
     if (savedAlarmConfigStr) {
       const savedAlarmConfig: { [key: string]: string } = JSON.parse(
         savedAlarmConfigStr
       );
       Object.keys(savedAlarmConfig).forEach(key => {
         if (savedAlarmConfig[key]) {
-          initFormFields[key].value = savedAlarmConfig[key];
+          setValue(key, savedAlarmConfig[key]);
         }
       });
     }
-    setFormFields(initFormFields);
-  };
-
-  /**
-   * This is used to validate form before saving.
-   * Each field object will be updated with the validity status and error messages will be displayed accordingly.
-   * If form is valid, configurations saving function will be called.
-   */
-  const validateForm = () => {
-    const updatedFormFields: FormFields = {};
-    const savingConfig: SavingConfig = {};
-    let formInvalid = false;
-    Object.keys(formFields).forEach(key => {
-      const fieldValue = formFields[key].value;
-      updatedFormFields[key] = {
-        ...formFields[key],
-        invalid: fieldValue === '' || fieldValue === undefined,
-      };
-      if (fieldValue === '' || fieldValue === undefined) {
-        formInvalid = true;
-      } else {
-        savingConfig[key] = fieldValue;
-      }
-    });
-    setFormFields(updatedFormFields);
-    if (!formInvalid) {
-      saveAlarmConfig(savingConfig);
-    }
   };
 
   /**
    * This function is used to save alarm configurations.
    * A alert will be displayed finally based on the saving status.
    */
-  const saveAlarmConfig = (savingConfig: SavingConfig) => {
+  const onSubmit = (data: FormField) => {
     let actionStatus = true;
     try {
-      localStorage.setItem(ALARM_DOC_NAME, JSON.stringify(savingConfig));
+      localStorage.setItem(ALARM_DOC_NAME, JSON.stringify(data));
       handleClose();
     } catch (Error) {
       actionStatus = false;
```

Comment:
you swallow the exception here. How would the caller react to exceptions?

Reply:
Based on the saving status, we show an alert whether saving is success or not